### DISCUSSION TITLE
How to 'manage' @rejected event  (q-upload) with ui.upload

### DISCUSSION THREAD

**Discussion by Derf000** [zauberzeug/nicegui#536] [Q&A] [answered]
Hi, 
I  am very new to nicegui  with some basic python skills but none in  javascript.
Reading the Nicegui doc, Nice gui element code  And quasar doc (QUploader <https://quasar.dev/vue-components/uploader>)
I was able to set the 'max-file-size' property to my "file uploader" project', 
But i can't find how to set/implement the OnRejected event 
If someone could show me a simple example, ...
Thanks by advance

Small code as example for 'my file uploader' :

```py
from nicegui import app, ui
from nicegui.events import UploadEventArguments

def handle_upload(e: UploadEventArguments) -> None:
    txt_name = e.name
    content = e.content
    my_type = e.type
    ui.notify(f'{txt_name} loaded')

...
g_file_uploader = ui.upload(label="Select - Drag file :  max 100kb", auto_upload=True, on_upload=handle_upload)\
  .props(' color="amber" text-color="black" max-file-size="102400" @rejected="onRejected" ')
...
```

**Comment by falkoschindler** [comment, marked as answer]
Good question!

Most NiceGUI elements only provide a very limited number of events as arguments, like `ui.button(on_click=...)` or `ui.upload(on_upload=...)`. But you can [register additional event handlers](https://nicegui.io/reference#generic_events) using the `.on()` method:

```py
def handle_upload(e: UploadEventArguments) -> None:
    ui.notify(f'{e.name} loaded')

ui.upload(label='Select file', auto_upload=True, on_upload=handle_upload) \
    .props('max-file-size=100') \
    .on('rejected', lambda: ui.notify('File rejected!'))
```

Of course you can pass either a lambda statement or a regular function. Optionally you can accept a `msg` dictionary to consume additional arguments:

```py
    .on('rejected', lambda msg: print(msg))
```

This needs some experimentation with the [Quasar documentation](https://quasar.dev/vue-components/uploader) at hand to find out about the dictionary content.

**Reply by Derf000** [reply]
Thanks for the link to the generic events documentation that I had missed.
It works perfectly, and it's exactly what I needed.
Thanks also for this very useful project.

**Comment by Derf000** [comment]
Thanks for adding the "on_rejected" event for the upload component.

Now i have a new question  about this  :  
how can i get the cause of the rejection (file size limit exceeded, total file size exceeded, number of files exceeded, ...)

More generally : is it possible to access of the properties (max_file_size for example) of the 'element' (upload)  through the sender or client objects  sended by an EventArguments when on_rejected is fired ?

**Reply by falkoschindler** [reply]
Currently there is no public API to get props. But you can access `._props['max_file_size']` for example.

**Reply by Derf000** [reply]
thank you for the reactivity
👍
